cmd/cmdexecutor/cmdexecutor.go

```diff
@@ -158,15 +161,21 @@ func main() {
 	for _, executor := range executors {
 		ns, name := executor.GetPod()
 		podKey := createPodStringFromNameAndNamespace(ns, name)
-		go func(errChan chan error, doneChan chan bool, execInst cmdexecutor.Executor) {
+		go func(errChan chan error, stdoutChan chan string, doneChan chan bool, execInst cmdexecutor.Executor) {
 			err := execInst.Wait(time.Duration(statusCheckTimeout) * time.Second)
+
 			if err != nil {
 				errChan <- err
 				return
```

Comment:
Send a default or empty string to stdoutchan even for error case, else the caller may get blocked in the channel.

Reply:
Ok, we are not reading from stdout channel in case of error. It makes sense then.